javascript/javascript1/week3/main.js

```diff
@@ -0,0 +1,171 @@
+// Item array removal
+const names = [
+  "Peter",
+  "Ahmad",
+  "Yana",
+  "kristina",
+  "Rasmus",
+  "Samuel",
+  "katrine",
+  "Tala",
+];
+const nameToRemove = "Ahmad";
+for (let i = 0; i < names.length; i++) {
+  if (names[i] === nameToRemove) {
+    names.splice(i, 1);
+    i--;
+  }
+}
+console.log(names); // ['Peter', 'Yana', 'kristina', 'Rasmus', 'Samuel', 'katrine', 'Tala']
+
+// When will we be there??
+function getTravelTime(travelInfo) {
+  let travelTimeResult = travelInfo.destinationDistance / travelInfo.speed;
+  let travelTimeResultInMinutes =
+    travelTimeResult - Math.trunc(travelTimeResult);
+  return (
+    Math.trunc(travelTimeResult) +
+    " hours " +
+    Math.trunc(travelTimeResultInMinutes * 60) +
+    " minutes"
+  );
+}
+const travelInformation = {
+  speed: 50,
+  destinationDistance: 432,
+};
+
+const travelTime = getTravelTime(travelInformation);
+console.log(travelTime); // 8 hours and 38 minutes
+
+// Series duration of my life
+const seriesDurations = [
+  {
+    title: "Game of thrones",
+    days: 3,
+    hours: 1,
+    minutes: 0,
+  },
+  {
+    title: "Sopranos",
+    days: 3,
+    hours: 14,
+    minutes: 0,
+  },
+  {
+    title: "The Wire",
+    days: 2,
+    hours: 12,
+    minutes: 0,
+  },
+  {
+    title: "Bones",
+    days: 7,
+    hours: 3,
+    minutes: 0,
+  },
+];
+const lifeDuration = 80 * 365 * 24 * 60;
+let totalPersent = 0;
+function logOutSeriesText() {
+  for (let i = 0; i < seriesDurations.length; i++) {
+    let perscent = Number(
+      (
+        ((seriesDurations[i].days * 24 * 60 +
+          seriesDurations[i].hours * 60 +
+          seriesDurations[i].minutes) /
+          lifeDuration) *
+        100
+      ).toFixed(3)
+    );
+
+    console.log(
+      seriesDurations[i].title + " tooks " + perscent + "% of my life"
+    );
+    totalPersent += perscent;
+  }
+}
+logOutSeriesText();
+console.log("In total that is " + totalPersent + "% of my life");
+
+// Smart-ease - Back to the basics!
+let notes = [];
+function saveNote(content, id) {
+  notes.push({ content: content, id: id });
+}
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+console.log(notes);
+
+function getNote(id) {
+  for (let i = 0; i < notes.length; i++) {
+    if (notes[i].id === id) {
+      return notes[i];
+    }
+    return false;
+  }
+}
+
+const firstNote = getNote(3);
```

Comment:
this doest actually work here. well, it only works for the first note. so if you do getNote(2) you will get 'False'
The reason is on line 103 :) - what is happening is that you are not really going through the notes, but only checking the first one. if that is false, your function returns false.
The way to get around this is to remove the return from the if else and instead first if you initialize a temporary id variable with a value that cannot be an index (e.g. -1). Then you use this variable to stores the id of the note when found.  Finally, following your for loop, you can check if the value of the variable is changed or not. If it is still -1 it means nothing was found and you can return false, else you can return the note corresponding to that index.
I hope it makes sense :)